pkg/controller/clusterserviceversion/clusterserviceversion_controller.go

```diff
@@ -0,0 +1,160 @@
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_olm_clusterserviceversion_watcher")
+
+const operatorTag = "marketplace.redhat.com/operator"
+const watchTag = "razee/watch-resource"
+const allnamespaceTag = "olm.copiedFrom"
+
+// Add creates a new ClusterServiceVersion Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileClusterServiceVersion{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("clusterserviceversion-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	labelPreds := []predicate.Predicate{
+		predicate.Funcs{
+			UpdateFunc: func(evt event.UpdateEvent) bool {
+				return false
+			},
+			CreateFunc: func(evt event.CreateEvent) bool {
+				if _, ok := evt.Meta.GetLabels()[allnamespaceTag]; ok {
+					return false
+				}
+				return true
+			},
+			GenericFunc: func(evt event.GenericEvent) bool {
+				return false
+			},
+		},
+	}
+
+	// Watch for changes to primary resource ClusterServiceVersion
+	err = c.Watch(&source.Kind{Type: &olmv1alpha1.ClusterServiceVersion{}}, &handler.EnqueueRequestForObject{}, labelPreds...)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// blank assignment to verify that ReconcileClusterServiceVersion implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileClusterServiceVersion{}
+
+// ReconcileClusterServiceVersion reconciles a ClusterServiceVersion object
+type ReconcileClusterServiceVersion struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a ClusterServiceVersion object and makes changes based on the state read
+// and what is in the ClusterServiceVersion.Spec
+func (r *ReconcileClusterServiceVersion) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Name", request.Name)
+	reqLogger.Info("Reconciling ClusterServiceVersion")
+	// Fetch the ClusterServiceVersion instance
+	CSV := &olmv1alpha1.ClusterServiceVersion{}
```

Comment:
```suggestion
	csv := &olmv1alpha1.ClusterServiceVersion{}
```
Style, should be lower case.